relayer/relayer.go

```diff
@@ -70,13 +72,37 @@ func (r *Relayer) Start(ctx context.Context) error {
 			r.logger.Error(ErrAttestationNotFound.Error())
 			continue
 		}
-		err = r.ProcessAttestation(ctx, att)
+
+		opts, err := r.EVMClient.NewTransactionOpts(ctx)
 		if err != nil {
 			r.logger.Error(ErrAttestationNotFound.Error())
 			time.Sleep(10 * time.Second)
 			// will keep retrying indefinitely
 			continue
 		}
+
+		tx, err := r.ProcessAttestation(ctx, opts, att)
+		if err != nil {
+			r.logger.Error(ErrAttestationNotFound.Error())
+			time.Sleep(10 * time.Second)
+			// will keep retrying indefinitely
+			continue
+		}
+
+		// wait for transaction to be mined
+		ethClient, err := r.EVMClient.NewEthClient()
+		if err != nil {
+			r.logger.Error(err.Error())
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		_, err = r.EVMClient.WaitForTransaction(ctx, ethClient, tx)
+		if err != nil {
+			r.logger.Error(err.Error())
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		ethClient.Close()
```

Comment:
[blocking question]
should this close be deferred? it looks like we could end up spinning up many of these, should we only be spinning up 1?

Reply:
> should this close be deferred?

I prefered to close it everytime not to leave hanging rpc connections. This is because we don't know how much time there will be between every attestation relay.

For example, now with the default 400block and 15sec block time, data commitments will be relayed every 1hour and a half. For valsets, that depends on the network.

That's why I chose to use the connection then close it immediatly.

> it looks like we could end up spinning up many of these, should we only be spinning up 1?

Only one will be spinned up because relaying attestations is sequential, there is no way to parallelize it.

Reply:
okay cool, good to know

> I prefered to close it everytime not to leave hanging rpc connections.

yeah that's what I was worried about with potentially spinning one up in each loop. why not only open a single one before we start the loop and then defer to close? atm, it look like we will spin up many if we receive an error on line 99

Reply:
> I prefered to close it everytime not to leave hanging rpc connections.

opening once before the loop starts and defering the close should guarantee that we do not do this, and as a bonus, its very flexible for future changes, and follows the standard practices of go

Reply:
got it :+1: change incoming

Reply:
[3219f45](https://github.com/celestiaorg/orchestrator-relayer/pull/85/commits/3219f4539469e97de43cb98c744ae8518c77192c)